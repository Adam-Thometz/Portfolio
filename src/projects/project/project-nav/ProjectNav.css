.ProjectNav {
  display: flex;
  justify-content: space-around;
  margin: 0 auto;
  width: 100%;
  border: none;
  font-size: 2rem;
}

.ProjectNav button {
  border: none;
  background-color: rgba(0,0,0,0);
  color: white;
  cursor: pointer;
  font-family: 'Nunito', sans-serif;
  transform: none;
  transition: transform 0.3s;
}

.ProjectNav button:hover {
  transform: scale(1.3);
}

.ProjectNav svg {
  transform: scale(1.3);
}

.ProjectNav svg:hover {
  transform: scale(1.6);
}