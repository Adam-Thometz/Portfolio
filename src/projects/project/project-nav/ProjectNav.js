import React from 'react';

import './ProjectNav.css';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft, faArrowRight } from '@fortawesome/free-solid-svg-icons';
import projects from '../../projectInfo';

const ProjectNav = ({ currProject, setShownProject }) => {
  const exit = () => setShownProject(null);
  const move = e => {
    let projectIdx = projects.findIndex(p => p.name === currProject);
    projectIdx += +e.currentTarget.id;
    if (projectIdx < 0) projectIdx = projects.length-1;
    if (projectIdx >= projects.length) projectIdx = 0;
    
    const nextProject = projects[projectIdx];
    setShownProject(nextProject);
  };

  return (
    <nav className='ProjectNav'>
        <button id="-1" title="Previous" onClick={move}><FontAwesomeIcon icon={faArrowLeft} />
        </button>
        <button className="Project-exit" onClick={exit}>Back to projects</button>
        <button id="1" title="Next" onClick={move}><FontAwesomeIcon icon={faArrowRight} />
        </button>
    </nav>
  );
};

export default ProjectNav;