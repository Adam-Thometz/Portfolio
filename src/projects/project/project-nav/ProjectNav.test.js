import React from "react";
import { render } from "@testing-library/react";
import ProjectNav from "./ProjectNav";

describe("ProjectNav component", () => {
  it("renders without crashing", () => {
    render(<ProjectNav />);
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<ProjectNav />);
    expect(asFragment()).toMatchSnapshot();
  });
});
