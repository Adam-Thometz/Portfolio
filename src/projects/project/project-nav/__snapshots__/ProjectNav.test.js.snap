// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectNav component matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="ProjectNav"
  >
    <button
      id="-1"
      title="Previous"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left fa-w-14 "
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="Project-exit"
    >
      Back to projects
    </button>
    <button
      id="1"
      title="Next"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-right fa-w-14 "
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          fill="currentColor"
        />
      </svg>
    </button>
  </nav>
</DocumentFragment>
`;
